src/core/Akka.Remote.Tests.Performance/InboundMessageDispatcherSpec.cs

```diff
@@ -32,9 +32,11 @@ private class BenchmarkActorRef : MinimalActorRef
         {
             private readonly Counter _counter;
 
-            public BenchmarkActorRef(Counter counter)
+            public BenchmarkActorRef(Counter counter, RemoteActorRefProvider provider)
```

Comment:
Fixing some bugs here that I introduced when I changed the way `IInboundMessageDispatcher` works - specifically, I changed the way two `IActorRef`s compare equality. Instead of using `==` they now use `.Equals`, which is correct. Unfortunately that caused this code to throw a `NullReferenceException` during the performance test due to the `Path` property of `BenchmarkActorRef` being `null`. These changes fix that.